### PR TITLE
Configure Webpack to create Sentry release and tag it in the Frontend

### LINKED ISSUE #413: Have Sentry know about git commit
There is a way to get sentry set up so that when there is a deploy and new bugs start happening, sentry will automatically detect the likely commit.

A lead is that vercel has this env variable:
VERCEL_GIT_COMMIT_SHA

It may involve something with the deploy hooks as well.

### PR DESCRIPTION
**Pull request by topocount**
Along with enabling integrations in GitHub and Vercel, this change
allows for commits to be associated with deploys in sentry.io.

In local environments, including `vercel dev`, the webpack plugin
defaults to "dry run" mode and doesn't attempt to create a release or
associate commits in Sentry. Commits are only tied to a release in
the specified config when vercel deploys from the "production" branch
in this configuration.

Additional configuration is below.

Closes #413 

test plan:

The "production" branch head in https://github.com/topocount/coordinape
is currently throwing an error on pageload. That error is associated
with the tagged release (represented by the commit SHA) which is
associated with recent commits.

## Vercel Integration with Sentry

[These instructions](https://docs.sentry.io/product/integrations/deployment/vercel/) up to but excluding "Usage" will suffice. The required webpack config is in this PR.

## Github Integration with Sentry

- [x] Steps 1-3 in [these instructions](https://docs.sentry.io/product/releases/setup/release-automation/) are all that's necessary. Since we're deploying with Vercel and have this webpack config, a Github Action is not needed.

### CODE REVIEW

File: craco.config.js
Comment (topocount) [inline craco.config.js:38]: These fields might need to be modified depending on the `org` and `project` slugs are configured in Sentry

File: package.json
Comment (exrhizo) [inline package.json:124]: pin dependencies `--exact`

File: craco.config.js
Comment (exrhizo) [reply craco.config.js:38]: org is `coordinape`
project is `app`

**Review by exrhizo — COMMENTED** [review]
So then,

this isn't needed?

image


I think @crabsinger added the sentry integration to vercel, can we merge this and it will work?

@levity @crabsinger probably still need to do the github setup?

File: craco.config.js
Comment (levity) [inline craco.config.js:38]:
```suggestion
          project: 'app',
```

**Review by levity — APPROVED** [review]
LGTM once you make the dependency change that @exrhizo suggested (personally i don't mind non-exact deps but i guess we should be consistent)

**Comment by topocount** [comment]
> So then,
> 
> image this isn't needed? (because you did the webpack version of feeding these in?)

This is still needed because this step will generate a project-scoped auth token that is graanted to vercel. I think this is much cleaner than any one of the contributors or maintainers generating the auth token via the [integration platform](https://docs.sentry.io/product/integrations/integration-platform/), essentially a manual integration.

> # Next steps
> 
> What is still needed for merge?

It seems like this has all been done

- [ ] The next steps are just the config documentation I linked in the PR description. The Vercel-Sentry setup will be needed prior to merging this PR or Vercel builds will likely fail, since this is setup to attempt to communicate with Sentry when a `VERCEL_GIT_COMMIT_SHA` environment variable is present.

- [ ] The Sentry-Github integration will be needed for associated commits to begin showing up for each Release.

It looks like the mandatory vercel-Sentry config might be done to prevent breaking vercel deployments, but I've reified this list of todos just to be sure.

**Comment by crabsinger** [comment]
I'll work with @topocount to setup the configs on Sentry and Vercel.

**Comment by crabsinger** [comment]
Added the Sentry Env Vars to vercel.

**Pull request merged**
